auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -310,16 +312,51 @@ public TypeSpec createTypeAdapter(ClassName className, ClassName autoValueClassN
     return classBuilder.build();
   }
 
+  private static void buildParameterizedToken(MethodSpec.Builder constructor, TypeName typeArg,
+                                              List<TypeVariableName> typeParams, boolean first) {
+    if (!first) {
+      constructor.addCode(", ");
+    }
+    if (typeArg instanceof ParameterizedTypeName) {
+      ParameterizedTypeName paramTypeArg = (ParameterizedTypeName) typeArg;
+      constructor.addCode("$T.newParameterizedTypeWithOwner(null, $T.class", $Gson$Types.class, paramTypeArg.rawType);
```

Comment:
You depend on a version but your users might be using a newer (or even
older) version. That class could be removed in the next patch release. It
won't, but it could.

On Fri, Aug 12, 2016 at 10:24 AM Ryan Harter <notifications@github.com>
wrote:

> In
> auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java
> <https://github.com/rharter/auto-value-gson/pull/66#discussion_r74599214>:
>
> > @@ -310,16 +312,51 @@ public TypeSpec createTypeAdapter(ClassName className, ClassName autoValueClassN
> >      return classBuilder.build();
> >    }
> >
> > +  private static void buildParameterizedToken(MethodSpec.Builder constructor, TypeName typeArg,
> > +                                              List<TypeVariableName> typeParams, boolean first) {
> > +    if (!first) {
> > +      constructor.addCode(", ");
> > +    }
> > +    if (typeArg instanceof ParameterizedTypeName) {
> > +      ParameterizedTypeName paramTypeArg = (ParameterizedTypeName) typeArg;
> > +      constructor.addCode("$T.newParameterizedTypeWithOwner(null, $T.class", $Gson$Types.class, paramTypeArg.rawType);
>
> @swankjesse <https://github.com/swankjesse> Do you mind providing some
> input here? If someone is using Gson can we consider it safe to use the
> internal $Gson$Types class, since we depend on a specific version of Gson?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rharter/auto-value-gson/pull/66/files/036c1f4a9a5caa2bf43e8c31da491c300a3892fe#r74599214>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEEEeyNoF2r7W7gog6FapDyArUgd1lmks5qfIIbgaJpZM4JiIcT>
> .
>